ethereum/test/pyth.js

```diff
@@ -260,23 +295,85 @@ contract("Pyth", function () {
         let ts = 1647273460;
         let rawBatch1 = generateRawBatchAttestation(ts - 5, ts, 1337);
         let rawBatch2 = generateRawBatchAttestation(ts + 5, ts + 10, 1338);
         const receipt = await updatePriceFeeds(this.pythProxy, [rawBatch1, rawBatch2]);
-        expectEvent(receipt, 'PriceUpdate');
+        expectEvent(receipt, 'PriceFeedUpdate', {
+            fresh: true,
+        });
+        expectEvent(receipt, 'BatchPriceFeedUpdate', {
+            batchSize: '10',
+            freshPricesInBatch: '10',
+        });
+        expectEvent(receipt, 'UpdatePriceFeeds', {
+            batchCount: '2',
+        });
     });
 
     it("should attest price updates with multiple batches of wrong order", async function () {
         let ts = 1647273460;
         let rawBatch1 = generateRawBatchAttestation(ts - 5, ts, 1337);
         let rawBatch2 = generateRawBatchAttestation(ts + 5, ts + 10, 1338);
         const receipt = await updatePriceFeeds(this.pythProxy, [rawBatch2, rawBatch1]);
-        expectEvent(receipt, 'PriceUpdate');
+        expectEvent(receipt, 'PriceFeedUpdate', {
+            fresh: true,
+        });
+        expectEvent(receipt, 'PriceFeedUpdate', {
+            fresh: false,
+        });
+        expectEvent(receipt, 'BatchPriceFeedUpdate', {
+            batchSize: '10',
+            freshPricesInBatch: '10',
+        });
+        expectEvent(receipt, 'BatchPriceFeedUpdate', {
+            batchSize: '10',
+            freshPricesInBatch: '0',
+        });
+        expectEvent(receipt, 'UpdatePriceFeeds', {
+            batchCount: '2',
+        });
+    });
+
+    it("should attest price updates with when required fee is given", async function () {
+        // Check that the owner is the default account Truffle
+        // has configured for the network.
+        const accounts = await web3.eth.getAccounts();
+        const defaultAccount = accounts[0];
+        assert.equal(await this.pythProxy.owner(), defaultAccount);
+
+        // Check initial fee is zero
+        assert.equal(await this.pythProxy.singleUpdateFeeInWei(), 0);
+
+        // Set fee 
+        await this.pythProxy.updateSingleUpdateFeeInWei(10);
+        assert.equal(await this.pythProxy.singleUpdateFeeInWei(), 10);
+
+        let ts = 1647273460;
+        let rawBatch1 = generateRawBatchAttestation(ts - 5, ts, 1337);
+        let rawBatch2 = generateRawBatchAttestation(ts + 5, ts + 10, 1338);
+
+        // Updating price feed with less than
+        let feeInWei = await this.pythProxy.getUpdateFee(2);
+
+        // When a smaller fee is payed it reverts
+        await expectRevert(
+            updatePriceFeeds(this.pythProxy, [rawBatch1, rawBatch2], feeInWei - 1), 
+            insufficientFeeError
+        );
+
+        const receipt = await updatePriceFeeds(this.pythProxy, [rawBatch1, rawBatch2], feeInWei);
+        expectEvent(receipt, 'UpdatePriceFeeds', {
+            fee: feeInWei
+        });
```

Comment:
can you add a test of doing the update that validates that the contract balance is updated properly?

Reply:
Good idea. Will do.